tmt/steps/provision/__init__.py

```diff
@@ -1788,6 +1789,84 @@ def fetch_logs(
         for log in guest_logs:
             log.store(logger, dirpath, log.name)
 
+    def _construct_mkdtemp_command(
+        self,
+        prefix: Optional[str] = None,
+        template: Optional[str] = None,
+        parent: Optional[Path] = None,
+    ) -> Command:
+        template = template or 'tmp.XXXXXXXXXX'
+
+        if prefix is not None:
+            template = f'{prefix}{template}'
+
+        options: list[str] = ['--directory']
+
+        if parent is not None:
+            options += ['-p', str(parent)]
```

Comment:
```suggestion
        if parent is not None:
            # Using --tmpdir instead of deprecated -p
            options += ['--tmpdir', str(parent)]
```
The manpage says `-p` is deprecated:
```
       -p DIR, --tmpdir[=DIR]
              interpret TEMPLATE relative to DIR; if DIR is not specified, use $TMPDIR if set, else  /tmp.   With  this
              option,  TEMPLATE  must not be an absolute name; unlike with -t, TEMPLATE may contain slashes, but mktemp
              creates only the final component

       -t     interpret TEMPLATE as a single file name component, relative to a directory: $TMPDIR, if  set;  else  the
              directory specified via -p; else /tmp [deprecated]
```

Reply:
That is true, and any modern `mktemp` will follow. But Alpine `mktemp` does not accept `--tmpdir`:

```
/ # mktemp --help
BusyBox v1.36.1 (2024-06-12 06:28:12 UTC) multi-call binary.

Usage: mktemp [-dt] [-p DIR] [TEMPLATE]

Create a temporary file with name based on TEMPLATE and print its name.
TEMPLATE must end with XXXXXX (e.g. [/dir/]nameXXXXXX).
Without TEMPLATE, -t tmp.XXXXXX is assumed.

        -d      Make directory, not file
        -q      Fail silently on errors
        -t      Prepend base directory name to TEMPLATE
        -p DIR  Use DIR as a base directory (implies -t)
        -u      Do not create anything; print a name

Base directory is: -p DIR, else $TMPDIR, else /tmp
```

I think even I started with `--tmpdir` or something similar, and Alpine tests did not work.